apps/challenges/views.py

```diff
@@ -523,9 +523,9 @@ def challenge_phase_detail(request, challenge_pk, pk):
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
     try:
-        challenge_phase = ChallengePhase.objects.get(pk=pk)
+        challenge_phase = ChallengePhase.objects.get(challenge=challenge, pk=pk)
     except ChallengePhase.DoesNotExist:
-        response_data = {"error": "ChallengePhase does not exist"}
+        response_data = {"error": "Challenge Phase {} does not exist".format(pk)}
```

Comment:
Please modify the error message to `{"error": "Challenge Phase {} does not exist for challenge".format(pk, challenge.pk)}`